libbeat/processors/decode_xml/decode_xml.go

```diff
@@ -0,0 +1,156 @@
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package decode_xml
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/v7/libbeat/common/enc/mxj"
+	"github.com/elastic/beats/v7/libbeat/common/jsontransform"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/beats/v7/libbeat/processors"
+	"github.com/elastic/beats/v7/libbeat/processors/checks"
+	jsprocessor "github.com/elastic/beats/v7/libbeat/processors/script/javascript/module/processor"
+)
+
+type decodeXML struct {
+	config decodeXMLConfig
+	log    *logp.Logger
+}
+
+var (
+	errFieldIsNotString = errors.New("field value is not a string")
+)
+
+const (
+	procName = "decode_xml"
+	logName  = "processor." + procName
+)
+
+func init() {
+	processors.RegisterPlugin(procName,
+		checks.ConfigChecked(New,
+			checks.RequireFields("fields"),
+			checks.AllowedFields("fields", "overwrite_keys", "add_error_key", "target", "document_id")))
+	jsprocessor.RegisterPlugin(procName, New)
+}
+
+// New construct a new decode_xml processor.
+func New(c *common.Config) (processors.Processor, error) {
+	config := defaultConfig()
+
+	if err := c.Unpack(&config); err != nil {
+		return nil, fmt.Errorf("fail to unpack the "+procName+" processor configuration: %s", err)
+	}
+
+	return newDecodeXML(config)
+
+}
+
+func newDecodeXML(config decodeXMLConfig) (processors.Processor, error) {
+	cfgwarn.Experimental("The " + procName + " processor is experimental.")
+
+	log := logp.NewLogger(logName)
+
+	return &decodeXML{config: config, log: log}, nil
+
+}
+
+func (x *decodeXML) Run(event *beat.Event) (*beat.Event, error) {
+	var errs []error
+	var field = x.config.Field
+	data, err := event.GetValue(field)
+	if err != nil {
+		if x.config.IgnoreMissing && err == common.ErrKeyNotFound {
+			return event, nil
+		}
+		errs = append(errs, err)
+	}
+	text, ok := data.(string)
+	if !ok {
+		errs = append(errs, errFieldIsNotString)
+	} else {
+		xmloutput, err := x.decodeField(text)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to decode fields in decode_xml processor: %v", err))
+		}
+
+		target := field
```

Comment:
This logic could be move to the constructor such that `Target` is set once based on the behavior you define.